File: 06/Java/TripleExpression/Abs.java
package TripleExpression;

public class Abs implements TripleExpression {
	private TripleExpression value;
    public Abs(TripleExpression val) {
        value = val;
    }

    public int evaluate(int x, int y, int z) {
        return Math.abs(value.evaluate(x, y, z));
    }
}

File: 06/CSharp/TripleExpressionApp/TripleExpressionApp/Abs.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TripleExpressionApp
{
    public class Abs : ITripleExpression
    {
        private ITripleExpression value;
        public Abs(ITripleExpression value)
        {
            this.value = value;
        }

        public double Evaluate(double x, double y, double z)
        {
            return Math.Abs(value.Evaluate(x, y, z));
        }
    }
}
